kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -604,21 +617,43 @@ public void start(Map<String, String> properties) {
     enableRetries = config.getBoolean(BigQuerySinkConfig.ENABLE_RETRIES_CONFIG);
   }
 
-  private void initializeStorageApiBasicMode() {
-    if(testStorageWriteApi != null) {
-      logger.info("Starting task with Test Storage Write API Default Stream...");
+  private void initializeStorageApiMode() {
+    if (testStorageWriteApi != null) {
+      logger.info("Starting task with Test Storage Write API Stream...");
       storageApiWriter = testStorageWriteApi;
+      batchHandler = testStorageApiBatchHandler;
+      loadExecutor = Executors.newScheduledThreadPool(1);
```

Comment:
What is the reason behind having a value of 1 here?

Reply:
We want to only have 1 thread committing streams.